elastic/apm-server
# intake: add cloud.service.name to metadata

## Motivation/summary
This PR adds `cloud.service.name` to the set of available cloud information in the `metadata`. The field was introduced with https://github.com/elastic/ecs/pull/1204.


## Checklist



- [x] Documentation has been updated (automatically)

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes
Send a `metadata.cloud.service.name` information along with event data and check that it is indexed as `keyword` for `error,transaction,span,metric` events. 


## Related issues
closes 4625


## How to add a new Intake API Field and index it
* Add the new field to the [modeldecoder](https://github.com/elastic/apm-server/pull/4626/files#diff-e1fc2ee084f735532c843f42168cb9e4ec49ba1f76225c6bf28aa261737f28a1). In case the field should also be available for the new RUM API, also add it to the [rumv3/model](https://github.com/elastic/apm-server/blob/master/model/modeldecoder/rumv3/model.go)
* Add the new field to the [model](https://github.com/elastic/apm-server/pull/4626/files#diff-41a73cb3dbc928cc6f439d7437ce8b5a5f099e333314cd1188c825e72713e55c)
* Add the [mapping from the modeldecoder to the model](https://github.com/elastic/apm-server/pull/4626/files#diff-72e4b347bf8ae6903277fb912273517840f3092e9720e7d6c32c32fc3dd64dc3)
* Add the new field to the testdata and update the generated test `approval` files by running `go test ./...` and `make check-approvals`
* Add the field names to the [meta/fields.yml](https://github.com/elastic/apm-server/pull/4626/files#diff-831985d9e9ab2fa8d8306e14130efbdba8a1e8ffa40507740a3dcb8469e5a2cc) where the new fields should be indexed. 
* Run `make update` and ensure that the [Intake Spec](https://github.com/elastic/apm-server/tree/master/docs/spec), the [generated model logic](https://github.com/elastic/apm-server/blob/master/model/modeldecoder/v2/model_generated.go) and the [apmpackage](https://github.com/elastic/apm-server/tree/master/apmpackage/apm/0.1.0) and the generated fields are properly updated
